### ISSUE TITLE
IVF-Flat sample filter is not applied in all required places

### ISSUE THREAD

**Issue by achirkin** [rapidsai/raft#1620]
It seems like the filtering in ivf-flat is not entirely correct, it's only applied to one if case out of the few.
Perhaps, it makes sense to shift the condition further below to the place where the found value is added to the queue.

https://github.com/rapidsai/raft/blob/1034a41e5dd956c1a132b41cb1e12a42ba7b2b38/cpp/include/raft/neighbors/detail/ivf_flat_interleaved_scan-inl.cuh#L740-L768

**Comment by lowener** [comment]
The call to `sample_filter` is not right.
It should be `sample_filter(queries_offset + blockIdx.y, list_id, vec_id)`.

**Comment by lowener** [comment]
True, the filtering was still adding neighbors that didn't pass the test to the queue. I am fixing this in PR #1831
